src/traces/table/plot.js

```diff
@@ -0,0 +1,892 @@
+        .call(d3.behavior.drag()
+            .origin(function(d) {
+                d3.event.stopPropagation();
+                return d;
+            })
+            .on('drag', makeDragRow(gd, tableControlView, -1))
+            .on('dragend', function() {
+                // fixme emit plotly notification
+            })
+        );
+
+    // initial rendering: header is rendered first, as it may may have async LaTeX (show header first)
+    // but blocks are _entered_ the way they are due to painter's algo (header on top)
+    renderColumnCellTree(gd, tableControlView, columnBlock.filter(headerBlock), columnBlock);
+    renderColumnCellTree(gd, tableControlView, columnBlock.filter(cellsBlock), columnBlock);
+
+    var scrollAreaClip = tableControlView.selectAll('.scrollAreaClip')
+        .data(gup.repeat, gup.keyFun);
+
+    scrollAreaClip.enter()
+        .append(c.clipView ? 'g' : 'clipPath')
+        .classed('scrollAreaClip', true)
+        .attr('id', function(d) { return 'scrollAreaBottomClip_' + d.key;});
+
+    var scrollAreaClipRect = scrollAreaClip.selectAll('.scrollAreaClipRect')
+        .data(gup.repeat, gup.keyFun);
+
+    scrollAreaClipRect.enter()
+        .append('rect')
+        .classed('scrollAreaClipRect', true)
+        .attr('x', -c.overdrag)
+        .attr('y', -c.uplift)
+        .attr('stroke', 'orange')
+        .attr('stroke-width', 2)
+        .attr('fill', 'none')
+        .style('pointer-events', 'stroke');
+
+    scrollAreaClipRect
+        .attr('width', function(d) {return d.width + 2 * c.overdrag;})
+        .attr('height', function(d) {return d.height + c.uplift;});
+
+    var columnBoundary = yColumn.selectAll('.columnBoundary')
+        .data(gup.repeat, gup.keyFun);
+
+    columnBoundary.enter()
+        .append('g')
+        .classed('columnBoundary', true);
+
+    var columnBoundaryClippath = yColumn.selectAll('.columnBoundaryClippath')
+        .data(gup.repeat, gup.keyFun);
+
+    // SVG spec doesn't mandate wrapping into a <defs> and doesn't seem to cause a speed difference
+    columnBoundaryClippath.enter()
+        .append(c.clipView ? 'g' : 'clipPath')
+        .classed('columnBoundaryClippath', true);
+
+    columnBoundaryClippath
+        .attr('id', function(d) {return 'columnBoundaryClippath_' + d.specIndex;});
+
+    var columnBoundaryRect = columnBoundaryClippath.selectAll('.columnBoundaryRect')
+        .data(gup.repeat, gup.keyFun);
+
+    columnBoundaryRect.enter()
+        .append('rect')
+        .classed('columnBoundaryRect', true)
+        .attr('fill', 'none')
+        .attr('stroke', 'magenta')
+        .attr('stroke-width', 2)
+        .style('pointer-events', 'stroke');
+
+    columnBoundaryRect
+        .attr('width', function(d) {return d.columnWidth;})
+        .attr('height', function(d) {return d.calcdata.height + c.uplift;});
+
+    updateBlockYPosition(null, cellsColumnBlock, tableControlView);
+};
+
+function renderScrollbarKit(tableControlView, gd) {
```

Comment:
Another scroller. That's now three separate implementations (legend, update menus and this here). :rage:

Reply:
Agree with your sentiment... I looked into these scrollers and didn't find an easy way for either using them or reworking them to cover the `table` use case, it's an infinite revolver scroller, our discussion: https://github.com/plotly/plotly.js/pull/1834#issuecomment-313725969

As the scrollbars and event captures didn't take much time and there were more difficult issues, I spent < 1 day putting in the scrollbars remembering that maybe you or us would check how unification could be done. I'd have spent a lot more time trying to unify while the core of `table` also wasn't stable so it was a risk reduction tradeoff (maybe a bad tradeoff on my part).

Reply:
That :rage: wasn't meant for you. Adding a separate scroller for `table` at this stage was definitively the correct decision.

It's just one big tech-debt that we'll have to pay back soon (I hope) :moneybag:

Reply:
Separate item: https://github.com/plotly/plotly.js/issues/2054